aiida/cmdline/commands/cmd_import.py

```diff
@@ -347,9 +189,210 @@ def cmd_import(
+    :param temp_folder: SandboxFolder, where the migrated file will be temporarily outputted.
+    :param file_to_import: Absolute path, including filename, of file to be migrated.
+    :param archive: Filename of archive to be migrated, and later attempted imported.
+    :param non_interactive: Whether or not the user should be asked for input for any reason.
+    :param more_archives: Whether or not there are more archives to be imported.
+    :param silent: Suppress console messages.
+    :return: Absolute path to migrated archive within SandboxFolder.
+    """
+    from aiida.common.log import override_log_formatter_context
+    from aiida.tools.importexport import detect_archive_type, EXPORT_VERSION
+    from aiida.tools.importexport.archive.migrators import get_migrator
+
+    # Echo start
+    echo.echo_info(f'migrating archive {archive}')
+
+    # Initialization
+    temp_out_file = 'migrated_importfile.aiida'
+
+    # Migration
+    try:
+        migrator_cls = get_migrator(detect_archive_type(file_to_import))
+        migrator = migrator_cls(file_to_import)
+        with override_log_formatter_context('%(message)s'):
+            migrator.migrate(EXPORT_VERSION, temp_folder.get_abs_path(temp_out_file), out_compression='zip')
+    except Exception as exception:
+        _echo_error(
+            f'an exception occurred while migrating the archive: {archive}.\n'
+            "Use 'verdi export migrate' to update this archive file.",
+            non_interactive=non_interactive,
+            more_archives=more_archives,
+            raised_exception=exception
+        )
+    else:
+        # Success
+        echo.echo_info('proceeding with import')
+
+        return temp_folder.get_abs_path(temp_out_file)
+
+
+def _echo_error(  # pylint: disable=unused-argument
+    message, non_interactive, more_archives, raised_exception, **kwargs
+):
+    """Utility function to help write an error message for ``verdi import``
+
+    :param message: Message following red-colored, bold "Error:".
+    :type message: str
+    :param non_interactive: Whether or not the user should be asked for input for any reason.
+    :type non_interactive: bool
+    :param more_archives: Whether or not there are more archives to import.
+    :type more_archives: bool
+    :param raised_exception: Exception raised during error.
+    :type raised_exception: `Exception`
+    """
+    from aiida.tools.importexport import IMPORT_LOGGER
+
+    IMPORT_LOGGER.debug('%s', traceback.format_exc())
+
+    exception = f'{raised_exception.__class__.__name__}: {str(raised_exception)}'
+
+    echo.echo_error(message)
+    echo.echo(exception)
+
+    if more_archives:
+        # There are more archives to go through
+        if non_interactive:
+            # Continue to next archive
+            pass
+        else:
+            # Ask if one should continue to next archive
+            click.confirm('Do you want to continue?', abort=True)
```

Comment:
well yeh I didn't decide on the logic to allow for failures, it was already there and I didn't want to change the behaviour.
Happy to remove it though

Reply:
Up to you

Reply:
I've simplified the cmd_import in ffb3699.
As the commit message says, I also removed the click.confirm for migrations, since this is already controlled by --migration/--no-migration

This is the output of a migrated import:

```
$ verdi import -v DEBUG tmp/mount_folder/tests/static/export/migrate/export_v0.2_simple.aiida
Calling import function import_data_dj for the django backend.
Warning: initial import of tmp/mount_folder/tests/static/export/migrate/export_v0.2_simple.aiida failed, trying migration: IncompatibleArchiveVersionError: Archive file version is 0.2, can read only version 0.9
Use 'verdi export migrate' to update this archive file.
Migration pathway: 0.2 -> 0.3 -> 0.4 -> 0.5 -> 0.6 -> 0.7 -> 0.8 -> 0.9
Extracting archive to temporary folder
Extracting zip files                     100.0%|██████████████████████████████████████████████████████████| 76/76
Performing migrations: 0.8 -> 0.9        100.0%|████████████████████████████████████████████████████████████| 7/7
Re-compressing archive as 'zip'
Moving archive to: /home/aiida/.aiida/repository/other3/sandbox/tmpo644hzqn/migrated_archive.zip
Info: proceeding with import
Calling import function import_data_dj for the django backend.

IMPORT
--------  --------------------
Archive   migrated_archive.zip

Parameters
--------------------------  ------
Comment rules               newest
New Node Extras rules       import
Existing Node Extras rules  kcl
CHECKING IF NODES FROM LINKS ARE IN DB OR ARCHIVE...
CREATING PK-2-UUID/EMAIL MAPPING...
Importing 16 entities
ASSESSING IMPORT DATA...
Finding existing entities - User         100.0%|████████████████████████████████████████████████████████████| 1/1
Reading archived entities - User         100.0%|████████████████████████████████████████████████████████████| 1/1
Finding existing entities - Computer     100.0%|████████████████████████████████████████████████████████████| 1/1
Reading archived entities - Computer     100.0%|████████████████████████████████████████████████████████████| 1/1
Finding existing entities - Node         100.0%|██████████████████████████████████████████████████████████| 10/10
Reading archived entities - Node         100.0%|██████████████████████████████████████████████████████████| 10/10
Finding existing entities - Group        100.0%|████████████████████████████████████████████████████████████| 4/4
Reading archived entities - Group        100.0%|████████████████████████████████████████████████████████████| 4/4
STORING ENTITIES...
Users -  existing entries                100.0%|████████████████████████████████████████████████████████████| 1/1
Computers -  existing entries            100.0%|████████████████████████████████████████████████████████████| 1/1
Nodes -  existing entries                100.0%|██████████████████████████████████████████████████████████| 10/10
Nodes - UUID=f75aaf40 - Extras           100.0%|██████████████████████████████████████████████████████████| 10/10
Groups -  existing entries               100.0%|████████████████████████████████████████████████████████████| 4/4
STORING NODE LINKS...
Links - label=output_array               100.0%|████████████████████████████████████████████████████████████| 8/8
   (0 new links...)
STORING GROUP ELEMENTS...
Groups -                                 100.0%|████████████████████████████████████████████████████████████| 4/4
Done (cleaning up)                       100.0%|██████████████████████████████████████████████████████████| 10/10

Summary
-----------------------  ---------------
Auto-import Group label  20201031-082514
User(s)                  1 existing
Computer(s)              1 existing
Node(s)                  10 existing
Group(s)                 4 existing

Success: imported archive tmp/mount_folder/tests/static/export/migrate/export_v0.2_simple.aiida
```

or with --no-migration

```
$ verdi import -v DEBUG tmp/mount_folder/tests/static/export/migrate/export_v0.2_simple.aiida --no-migration
Calling import function import_data_dj for the django backend.
Traceback (most recent call last):
  File "/tmp/mount_folder/aiida/cmdline/commands/cmd_import.py", line 220, in _import_archive
    import_data(archive_path, **import_kwargs)
  File "/tmp/mount_folder/aiida/tools/importexport/dbimport/__init__.py", line 87, in import_data
    return import_data_dj(in_path, group=group, **kwargs)
  File "/tmp/mount_folder/aiida/tools/importexport/dbimport/backends/django.py", line 128, in import_data_dj
    reader.check_version()
  File "/tmp/mount_folder/aiida/tools/importexport/archive/readers.py", line 140, in check_version
    raise IncompatibleArchiveVersionError(msg)
aiida.tools.importexport.common.exceptions.IncompatibleArchiveVersionError: Archive file version is 0.2, can read only version 0.9
Use 'verdi export migrate' to update this archive file.

Critical: an exception occurred while importing the archive tmp/mount_folder/tests/static/export/migrate/export_v0.2_simple.aiida: IncompatibleArchiveVersionError: Archive file version is 0.2, can read only version 0.9
Use 'verdi export migrate' to update this archive file.
```

Reply:
should also check manually with a webpage. What a good one to try?

Reply:
> I've simplified the cmd_import in [ffb3699](https://github.com/aiidateam/aiida-core/commit/ffb36990ec9069b8f078be2eb20dd6aece2193b7).

Thanks!


> I also removed the click.confirm for migrations, since this is already controlled by --migration/--no-migration

For me that change is fine, please also update the documentation (search for "prompt" in share_data.rst


> should also check manually with a webpage. Whats a good one to try?

One of the files from https://archive.materialscloud.org/record/2018.0001/v1 ?

Reply:
yep that worked fine 👍 

```
$ verdi import -w https://archive.materialscloud.org/record/2018.0001/v1Info: retrieving archive URLS from https://archive.materialscloud.org/record/2018.0001/v1
Success: 2 archive URLs discovered and added
Info: downloading archive https://archive.materialscloud.org/record/file?record_id=20&file_id=c8c80235-cf81-4660-b0cf-3ca5824c3725&filename=SSSP_efficiency_pseudos.aiida
Success: archive downloaded, proceeding with import
Warning: initial import of https://archive.materialscloud.org/record/file?record_id=20&file_id=c8c80235-cf81-4660-b0cf-3ca5824c3725&filename=SSSP_efficiency_pseudos.aiida failed, trying migration: IncompatibleArchiveVersionError: Archive file version is 0.3, can read only version 0.9
Use 'verdi export migrate' to update this archive file.
Migration pathway: 0.3 -> 0.4 -> 0.5 -> 0.6 -> 0.7 -> 0.8 -> 0.9
Extracting archive to temporary folder
Re-compressing archive as 'zip'                                                                                  
Moving archive to: /home/aiida/.aiida/repository/other3/sandbox/tmpubonj4bn/migrated_archive.zip
Info: proceeding with import

IMPORT
--------  --------------------
Archive   migrated_archive.zip

Parameters
--------------------------  ------
Comment rules               newest
New Node Extras rules       import
Existing Node Extras rules  kcl
                                                                                                                 
Summary
-----------------------  ---------------
Auto-import Group label  20201031-110237
User(s)                  1 new
Node(s)                  85 new
Group(s)                 1 new

Success: imported archive https://archive.materialscloud.org/record/file?record_id=20&file_id=c8c80235-cf81-4660-b0cf-3ca5824c3725&filename=SSSP_efficiency_pseudos.aiida
Info: downloading archive https://archive.materialscloud.org/record/file?record_id=20&file_id=bb240983-f808-4a0c-aba9-ce4a728820ac&filename=SSSP_accuracy_pseudos.aiida
Success: archive downloaded, proceeding with import
Warning: initial import of https://archive.materialscloud.org/record/file?record_id=20&file_id=bb240983-f808-4a0c-aba9-ce4a728820ac&filename=SSSP_accuracy_pseudos.aiida failed, trying migration: IncompatibleArchiveVersionError: Archive file version is 0.3, can read only version 0.9
Use 'verdi export migrate' to update this archive file.
Migration pathway: 0.3 -> 0.4 -> 0.5 -> 0.6 -> 0.7 -> 0.8 -> 0.9
Extracting archive to temporary folder
Re-compressing archive as 'zip'                                                                                  
Moving archive to: /home/aiida/.aiida/repository/other3/sandbox/tmpk18emw11/migrated_archive.zip
Info: proceeding with import

IMPORT
--------  --------------------
Archive   migrated_archive.zip

Parameters
--------------------------  ------
Comment rules               newest
New Node Extras rules       import
Existing Node Extras rules  kcl
                                                                                                                 
Summary
-----------------------  ---------------
Auto-import Group label  20201031-110248
User(s)                  1 existing
Node(s)                  11 new
Group(s)                 1 new

Success: imported archive https://archive.materialscloud.org/record/file?record_id=20&file_id=bb240983-f808-4a0c-aba9-ce4a728820ac&filename=SSSP_accuracy_pseudos.aiida
```

Reply:
and I've updated `share_data.rst`

Reply:
> I thought there was a test for archives imported from URLs as well... perhaps I remember incorrectly?

there may well be, I haven't looked to change any of them tests